client/src/pages/CodeUpload.js

```diff
@@ -128,13 +128,19 @@ const CodeUpload = () => {
         data: JSON.stringify(requestData)
       });
       //redirect user to their reviews page
-      if (response.data.success) {
+      const removeCredit = await axios.put(`/user/${user._id}/add-credit`, {
+        credits: -1
+      });
+      if (response.data.success && removeCredit.data.success) {
         var alerts = new Set();
```

Comment:
Should probably attempt to remove credits and only after that's successful, post the request.

Reply:
Fixed, also added error snackbar when user doesn't have enough credits.